ic-factory/src/api.rs

```diff
@@ -26,6 +26,7 @@ pub trait FactoryCanister: Canister + Sized + PreUpdate {
     /// otherwise, cycles balances of `principal` is returned.
     /// If `principal` does not exists, `None` is returned.
     #[update(trait = true)]
+    #[allow(clippy::needless_lifetimes)]
```

Comment:
Let's not have this. I think @awkure has fixed this so correct life time syntax can be used